sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/refactoring/strategy/Sqlite.kt

```diff
@@ -0,0 +1,59 @@
+package app.cash.sqldelight.intellij.refactoring.strategy
+
+class Sqlite : SqlGeneratorStrategy {
+
+  override fun tableNameChanged(oldName: String, newName: String): String {
+    return "ALTER TABLE $oldName RENAME TO $newName;"
+  }
+
+  override fun columnAdded(tableName: String, columnDef: String): String {
+    return "ALTER TABLE $tableName ADD COLUMN $columnDef;"
+  }
+
+  override fun columnRemoved(
+    tableName: String,
+    columnName: String,
+    columnDefList: List<String>
+  ): String {
+    val columnNames = columnDefList.filter { !it.startsWith(columnName) }
+      .joinToString(", ") { columnDef -> columnDef.takeWhile { !Character.isWhitespace(it) } }
+    val columnDefString = columnDefList.filter { !it.startsWith(columnName) }
+      .joinToString(", ")
+
+    return """
+      |BEGIN TRANSACTION;
```

Comment:
Is this something that changes with sqldelight 2.x? I was under the impression that at least for iOS and Android in 1.x world, you don't need transactions. There's also some documentation about it: https://cashapp.github.io/sqldelight/jvm_sqlite/migrations/

Reply:
yea true theres no need for it